pkg/validations/createvalidations/gitops.go

```diff
@@ -28,6 +41,30 @@ func ValidateGitOps(ctx context.Context, k validations.KubectlClient, cluster *t
 	if err != nil {
 		return fmt.Errorf("workload cluster gitOpsConfig is invalid: %v", err)
 	}
+
+	return err
+}
+
+func validateAuthenticationGitProvider(clusterSpec *cluster.Spec) error {
+	fluxConfig := clusterSpec.FluxConfig
+	if fluxConfig.Spec.Git == nil {
+		return nil
+	}
+
+	password, _ := git.ReadEnvVar(EksaGitPasswordTokenEnv)
+	privateKeyFilePath, _ := git.ReadEnvVar(EksaGitPrivateKeyTokenEnv)
+
+	if password == "" {
```

Comment:
you could unnest this whole block by doing
```go
if password != "" {
  return nil
}
```

Reply:
The reason we don't want to return here is because the user can specify both password and private key, in which case we want to validate values on them both.